libfreerdp/core/peer.c

```diff
@@ -567,16 +567,26 @@ static int peer_recv_callback(rdpTransport* transport, wStream* s, void* extra)
 			break;
 
 		case CONNECTION_STATE_LICENSING:
-			if (!license_send_valid_client_error_packet(rdp->license))
-			{
-				WLog_ERR(TAG,
-				         "peer_recv_callback: CONNECTION_STATE_LICENSING - license_send_valid_client_error_packet() fail");
+		{
+			LicenseCallbackResult res = client->LicenseCallback(client, s);
```

Comment:
Sure this callback is always set? The struct is exposed in `peer.h`, so a NULL check with an appropriate error message could be helpful.

Reply:
make sense